### PR TITLE
1993 consolidating duplicate errors

### LINKED ISSUE #1993: Consolidate duplicated Errors across packages
_Follow up of https://github.com/ecadlabs/taquito/issues/160_ as described in https://hackmd.io/@tca0T8LZT3au8LG6hSX20g/r1zTRTlfj_

Task 2: Consolidate duplicated Errors across packages (InvalidViewParameterError, InvalidAddressError, InvalidBlockHashError, InvalidHexStringError, InvalidMessageError, ...)

### PR DESCRIPTION
**Pull request by hui-an-yang**
closes #1993 

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [x] Your code builds cleanly without any errors or warnings
- [x] You have run the linter against the changes
- [x] You have added unit tests (if relevant/appropriate)
- [ ] You have added integration tests (if relevant/appropriate)
- [x] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [x] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [x] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [x] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

Consolidated common error classes `InvalidAddressError `, `InvalidBlockHashError `, `InvalidDerivationPathError `, `InvalidHexStringError `, `InvalidMessageError `, and `InvalidViewParameterError `, into @taquito/core package

### CODE REVIEW

File: packages/taquito-contracts-library/package.json [hunk 1]
```diff
@@ -68,6 +68,7 @@
     ]
   },
   "dependencies": {
+    "@taquito/core": "^16.1.2-beta-RC.0",
```
Comment (dsawali) [inline packages/taquito-contracts-library/package.json:71]: I think master is already at 16.1.2, could you merge master into this branch for the latest versions?

File: packages/taquito/src/contract/rpc-contract-provider.ts [hunk 2]
```diff
@@ -421,10 +421,10 @@ export class RpcContractProvider
    */
   async transferTicket(params: TransferTicketParams) {
     if (validateAddress(params.destination) !== ValidationResult.VALID) {
-      throw new InvalidAddressError(params.destination, 'param destination');
+      throw new InvalidAddressError(params.destination, 'params destination');
```
Comment (dsawali) [inline packages/taquito/src/contract/rpc-contract-provider.ts:424]: I notice the `errorDetail` param being passed as `params destination`

How would the error message look like when an `errorDetail` is passed?

File: packages/taquito-contracts-library/package.json (hunk 1 above)
Comment (hui-an-yang) [reply packages/taquito-contracts-library/package.json:71]: will do

File: packages/taquito/src/contract/rpc-contract-provider.ts (hunk 2 above)
Comment (hui-an-yang) [reply packages/taquito/src/contract/rpc-contract-provider.ts:424]: for now `Address "${address}" is invalid. params destination` 
I understand it's not ideal, am looking to make it `Address "${address}" is invalid. "INVALID_CHECKSUM" found in transferTicket params destination`
Will need a bigger undertaking to show why it's invalid, thinking to do that in task3

Comment (dsawali) [reply packages/taquito/src/contract/rpc-contract-provider.ts:424]: personally I find the `params destination` bit could be a bit misleading if people don't know which part of the code throws it. I think we can omit it for now. What do you think?

Comment (hui-an-yang) [reply packages/taquito/src/contract/rpc-contract-provider.ts:424]: Will this help? `Address "${address}" is invalid. in transferTicket params destination`?
Was added there cuz the params can have InvalidAddressError with params source or destination
But I don't mind omit it for now to revisit them in task3

Comment (dsawali) [reply packages/taquito/src/contract/rpc-contract-provider.ts]: I think displaying the PKH is clear enough to pinpoint which parameter field was invalid. 

And in cases where both source and destination are the same, the errorDetail would be a bit redundant.

Comment (hui-an-yang) [reply packages/taquito/src/contract/rpc-contract-provider.ts]: they're all removed

**Review by dsawali — APPROVED** [review]

**Pull request merged**